packages/vm/src/checksum.rs

```diff
@@ -1,25 +1,49 @@
+use std::convert::TryFrom;
+
 use sha2::{Digest, Sha256};
 
+use crate::errors::{make_cache_err, VmError};
+
 /// A SHA-256 checksum of a Wasm blob, used to identify a Wasm code.
 /// This must remain stable since this checksum is stored in the blockchain state.
 ///
 /// This is often referred to as "code ID" in go-cosmwasm, even if code ID
 /// usually refers to an auto-incrementing number.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Default)]
+// Note: do not remove Default, we need it in go-cosmwasm (for now)
```

Comment:
Hmm, a Default here smells. Why do we need it?

Reply:
From go-cosmwasm: `src/error.rs`:

```rust
pub fn handle_c_error<T, E>(r: Result<T, E>, errout: Option<&mut Buffer>) -> T
where
    T: Default,
    E: fmt::Display,
{
    match r {
        Ok(t) => {
            clear_error();
            t
        }
        Err(e) => {
            set_error(e.to_string(), errout);
            T::default()
        }
    }
}
```

We use this all over the place. I would rather not try to refactor all of this right now.
If you want to tackle that need of Default and then remove it from cosmwasm-vm later this week, I am happy with that.

Reply:
Alright!